### ISSUE TITLE
System.Uri constructor returns an object that does is not equal to the input

### ISSUE THREAD

**Issue by edoust** [dotnet/runtime#56482]
### Description

Create a `Uri` that represents a `MAPI` item

Example code:
```c#
var input = "mapi16:///{S-1-5-0000000}/mail@example.com/곯가가가공갫갡곤갘갖갯걆겹갹곓곌갂겥강걥겤걿갨가";
var uri = new Uri(input, UriKind.Absolute);
```

Expected:
`uri` should have the same value as `input`

Actual:
`uri` has this value: `mapi16:///{S-1-5-0000000}/mail@example.com/곯가가가공갫갡곤갘갖갯걆겹갹곓곌갂겥강걥겤걿갨가/{S-1-5-0000000}/mail@example.com/곯가가가공갫갡곤갘갖갯걆겹갹곓곌갂겥강걥겤걿갨가`

### Configuration

Code is running on .NET 5

### Regression?

This was working in both .NET Framework and UWP (.NET Core)

**Comment by MihaZupan** [comment]
Doesn't repro on `net48`, reproes on `netcoreapp2.1` and newer. Looks like a Framework => Core regression.

Simplified repro
```c#
const string UriString = "custom:///ü";
Console.WriteLine(new Uri(UriString, UriKind.Absolute).AbsoluteUri);
```

This looks like another manifestation of #1061 that was fixed by #36429 for 5.0 - just for custom Uri schemes.

**Comment by karelz** [comment]
Triage: We should fix it. Not a regression against 5.0 -> moving to 7.0.
